adapters/adview/adview.go

```diff
@@ -157,3 +177,41 @@ func getMediaTypeForImp(impID string, imps []openrtb2.Imp) (openrtb_ext.BidType,
 	}
 	return mediaType, nil
 }
+
+func getMediaTypeForBid(bid openrtb2.Bid, reqImps []openrtb2.Imp) (openrtb_ext.BidType, error) {
+	mediaType := openrtb_ext.BidTypeBanner
+	for _, imp := range reqImps {
+		if bid.ImpID == imp.ID {
+			if bid.Ext != nil {
+				var bidExt *adviewBidExt
+				err := json.Unmarshal(bid.Ext, &bidExt)
+				if err == nil {
+					return getBidType(bidExt)
+				} else {
+					return mediaType, err
+				}
+			}
+			return mediaType, nil
+		}
+	}
+	return mediaType, nil
+}
+
+// getBidType returns the bid type specified in the response bid.ext
+func getBidType(bidExt *adviewBidExt) (openrtb_ext.BidType, error) {
+	// setting "banner" as the default bid type
+	bidType := openrtb_ext.BidTypeBanner
+	if bidExt != nil {
+		switch bidExt.BidType {
```

Comment:
Please consider implementing a pattern where the adapter server sets the [MType](https://github.com/prebid/openrtb/blob/main/openrtb2/bid.go#L334) field in the response to accurately determine the media type for the impression.

Reply:
Sorry, do you mean in line 203, bidType set to banner default type is not suitable, so I should return error if no matched bidtype, right?
Or you mean I should set Mtype such as "seatBid.Bid[i].MType = openrtb2.MarkupBanner" in MakeBids()?
Thank you.

Reply:
@Sonali-More-Xandr So, you mean is I can use the following codes to get the bid type is enough?
func getMediaTypeForBid(bid openrtb2.Bid) (openrtb_ext.BidType, error) {
	switch bid.MType {
	case openrtb2.MarkupBanner:
		return openrtb_ext.BidTypeBanner, nil
	case openrtb2.MarkupVideo:
		return openrtb_ext.BidTypeVideo, nil
	case openrtb2.MarkupNative:
		return openrtb_ext.BidTypeNative, nil
	default:
		return "", fmt.Errorf("Unable to fetch mediaType in: %s", bid.ImpID)
	}
}